### DISCUSSION TITLE
[QUESTION] Use libopus to encode pcm audio into opus audio files cannot be played

### DISCUSSION THREAD

**Discussion by Origin-wu** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1762] [Q&A] [answered]
I use samples\opusSampleFrames\sample-***.opus to play audio files, or combine multiple frames of sample-000~618.opus into a file such as: sample-test.opus can also play. However, I use libopus to encode pcm audio into opus audio files and it cannot be played.

./opus_demo -e  audio  48000 2  input.pcm  output.opus
libopus 1.3.1
Encoding 48000 Hz input at 0.000 kb/s in auto bandwidth with 960-sample frames.
average bitrate:              96.930 kb/s
maximum bitrate:             194.400 kb/s
active bitrate:               97.313 kb/s
bitrate standard deviation:   14.546 kb/s

Is there a problem with the opus audio file obtained by the above method? Can you help me look at this issue?
If the above method does not work, is there any way to get the opus audio file with the same codec as in the sample?

**Comment by disa6302** [comment, marked as answer]
Hello,

While I do not have clarity on how the library does the conversion, I would recommend using a normal player to see if opus decoding the generated opus frames playback fine before trying out the frames with the SDK.
